internal/architecture/architecture_test.go

```diff
@@ -0,0 +1,106 @@
+package architecture
```

Comment:
If you don't use `defaultArchitecture`, can you make this `architecture_test` instead?

Reply:
It is used in this test case:
```
It("should return defaultArchitecture if .spec.cluster is nil", func() {
	archs, err := GetSSPArchs(&ssp.SSPSpec{})
	Expect(err).ToNot(HaveOccurred())
	Expect(archs).To(ConsistOf(defaultArchitecture))
})
```

Reply:
What if you construct `defaultArchitecture` again in this test case? I'd like to see decoupling of unit tests from internal implementation details (which is best ensured by putting the unit tests into a different pkg)